src/main/java/games/strategy/triplea/delegate/MustFightBattle.java

```diff
@@ -480,8 +480,7 @@ private void removeAirNoLongerInTerritory() {
     // remove any air units that were once in this attack, but have now
     // moved out of the territory
     // this is an ilegant way to handle this bug
-    final CompositeMatch<Unit> airNotInTerritory = new CompositeMatchAnd<>();
-    airNotInTerritory.add(Matches.unitIsInTerritory(m_battleSite).invert());
+    final Match<Unit> airNotInTerritory = Matches.unitIsInTerritory(m_battleSite).invert();
```

Comment:
Probably option 2. People tend to report the bugs but not the poorly named methods :)